spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -0,0 +1,194 @@
+                                           "Please specify fields for one");
+    }
+    List<String> fields = getFeatureList(inputSchema, featuresToInclude, featuresToExclude, predictionField);
+    for (String field : fields) {
+      Schema.Field inputField = inputSchema.getField(field);
+      Schema schema = inputField.getSchema();
+      Schema.Type features = schema.isNullableSimple() ? schema.getNonNullable().getType() : schema.getType();
+      if (!(features.equals(Schema.Type.INT) || features.equals(Schema.Type.LONG) ||
+        features.equals(Schema.Type.FLOAT) || features.equals(Schema.Type.DOUBLE))) {
+        throw new IllegalArgumentException(String.format("Field to classify must be continuous : int, double, " +
+                                                           "float, long but was %s.", features));
+      }
+      Preconditions.checkArgument(!features.equals(Schema.Type.NULL), "Field to classify must not be of type null");
+    }
+
+  }
+
+  /**
+   * Get the feature list of the features that have to be used for training/prediction depending on the
+   * featuresToInclude or featuresToInclude list.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training/prediction.
+   * @param featuresToExclude features to be excluded when training/predicting.
+   * @param predictionField   field containing the prediction values.
+   * @return feature list to be used for training/prediction.
+   */
+  public static List<String> getFeatureList(Schema inputSchema, @Nullable String featuresToInclude,
+                                            @Nullable String featuresToExclude, String predictionField) {
+    if (!Strings.isNullOrEmpty(featuresToExclude) && !Strings.isNullOrEmpty(featuresToInclude)) {
+      throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
+                                           "Please specify fields for one");
+    }
+    List<String> fields = new ArrayList<>();
+    List<Schema.Field> inputSchemaFields = inputSchema.getFields();
+
+    if (!Strings.isNullOrEmpty(featuresToInclude)) {
+      List<String> features = Lists.newArrayList(Splitter.on(',').trimResults().split(featuresToInclude));
+      for (String field : features) {
+        Schema.Field inputField = inputSchema.getField(field);
+        if (!field.equals(predictionField) && inputField != null) {
+          fields.add(field);
+        }
+      }
+    } else {
+      Set<String> excludeFeatures = new HashSet<>();
+      if (!Strings.isNullOrEmpty(featuresToExclude)) {
+        excludeFeatures.addAll(Lists.newArrayList(Splitter.on(',').trimResults().split(featuresToExclude)));
+      }
+      for (Schema.Field field : inputSchemaFields) {
+        String fieldName = field.getName();
+        if (!fieldName.equals(predictionField) && (excludeFeatures.isEmpty() || !excludeFeatures.contains(fieldName))) {
```

Comment:
The reason why you don't need to check `excludeFeatures.isEmpty()` is because, if it is empty, the `excludeFeatures.contains` will always return false, hence no field will be exclude (i.e. all fields, except the `predictionField` will be included). I believe that's the logic you want, right?